### PR TITLE
feat: Qwen-image 2.1 support (CORE-423)

### PR DESCRIPTION
**Pull request by kijai**
## Qwen-Image 2.1

Single-stream DiT, RGBA VAE, Qwen3-VL-8B text encoder.

- **Attention** is exactly block-causal without flex: the sequence splits into segments, so image
  blocks stay unmasked on the fast kernels and only text runs carry a mask.
- **Prefix cache** — text and reference tokens modulate from t=0, so their K/V are step-independent
  and cached for the run (~1.7x on edits). Stands down for block hooks; falls back to RAM, then
  recompute.
- **Text encoder** drops vision hidden states and hands their positions to the transformer, which
  splices VAE latents there, decoupling vision tokens from latent size. Without a VAE the states
  stay and the image conditions through the encoder alone.
- **VAE** reuses the Wan 2.2 module parametrized for single images. All new arguments default to the
  old values.
- **Nodes**: `TextEncodeQwenImage21` (positive, negative, and an empty latent on the reference grid,
  since any other size shifts the edit) and `QwenImage21Cache` to adjust the KV cache device/quantization.

One deliberate divergence from upstream: reference grids are offset half a token when their parity
differs from the target, avoiding the more drastic position drift between source and result images.

### CODE REVIEW

File: comfy/ldm/wan/model_animate2.py [hunk 1]
```diff
@@ -294,7 +295,7 @@ def forward_orig(self, x, t, context, clip_fea=None, freqs=None, freqs_pose=None
 
         cache = transformer_options.get("animate2_cache", None) if apply_pose else None
         if cache is not None:
-            cache.select(pose_latents)
+            cache.select(pose_latents[:1])  # keyed on batch element 0, so a cond batch size change mid-run stays valid
```
[the remark this answers is not in the document]
Comment (kijai) [reply comfy/ldm/wan/model_animate2.py:298]: Pre-existing and not really an issue.

File: comfy/text_encoders/qwen_image21.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+class QwenImage21Qwen3VLClipModel(comfy.text_encoders.qwen3vl.Qwen3VLClipModel):
+    def __init__(self, device="cpu", dtype=None, attention_mask=True, model_options={}):
+        super().__init__(device=device, dtype=dtype, attention_mask=attention_mask, model_options=model_options, model_type="qwen3vl_8b")
+        # last layer without the final RMSNorm: transformers 4.57 hidden_states[-1], which Qwen's results are tuned to (5.x norms it)
+        self.layer_norm_hidden_state = False
+        self.image_spans = []
+
+    def process_tokens(self, tokens, device):
+        embeds, attention_mask, num_tokens, embeds_info = super().process_tokens(tokens, device)
+        self.image_spans = [(e["index"], e["size"]) for e in embeds_info if e["type"] == "image"]
```
[the remark this answers is not in the document]
Comment (kijai) [reply comfy/text_encoders/qwen_image21.py:39]: Unreachable case in practice...

**Comment by mingyi456** [comment]
@kijai Hi, I just saw that it uses flex attention, would it make sense to add flex attention to ComfyUI? Have you tested it?

**Comment by kijai** [comment]
> @kijai Hi, I just saw that it uses flex attention, would it make sense to add flex attention to ComfyUI? Have you tested it?

It doesn't have to, this is implemented without it and can even use int8 attention just fine.

**Review by comfyanonymous — APPROVED** [review]
